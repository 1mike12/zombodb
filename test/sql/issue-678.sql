create table issue678
(
    id          integer                not null generated by default as identity,
    name        character varying(250) not null,
    description text,
    primary key (id),
    constraint unique_name unique (name)
);
create index idxissue678 on issue678 using zombodb ((issue678.*)) with (url = 'http://localhost:9200/');

insert into issue678 (name, description)

select 'test' || numb::text, 'test value ' || numb::text
from generate_series(1, 100) gs (numb);

explain (costs off) select * from issue678 as t where t ==> 'test' order by t.name, t.id limit 1;
select * from issue678 as t where t ==> 'test' order by t.name, t.id limit 1;

drop table issue678;