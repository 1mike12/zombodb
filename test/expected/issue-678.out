create table issue678
(
    id          integer                not null generated by default as identity,
    name        character varying(250) not null,
    description text,
    primary key (id),
    constraint unique_name unique (name)
);
create index idxissue678 on issue678 using zombodb ((issue678.*)) with (url = 'http://localhost:9200/');
insert into issue678 (name, description)
select 'test' || numb::text, 'test value ' || numb::text
from generate_series(1, 100) gs (numb);
explain (costs off) select * from issue678 as t where t ==> 'test' order by t.name, t.id limit 1;
                                   QUERY PLAN                                   
--------------------------------------------------------------------------------
 Limit
   ->  Incremental Sort
         Sort Key: name, id
         Presorted Key: name
         ->  Index Scan using unique_name on issue678 t
               Filter: (ctid ==> '{"query_string":{"query":"test"}}'::zdbquery)
(6 rows)

select * from issue678 as t where t ==> 'test' order by t.name, t.id limit 1;
 id | name  | description  
----+-------+--------------
  1 | test1 | test value 1
(1 row)

drop table issue678;
